pkg/pluginmanager/manager.go

```diff
@@ -1104,13 +1104,14 @@ func doInstallTestPlugin(p *discovery.Discovered, pluginPath, version string) er
 }
 
 func updatePluginInfoAndInitializePlugin(p *discovery.Discovered, plugin *cli.PluginInfo) error {
-	c, err := catalog.NewContextCatalog(p.ContextName)
+	c, err := catalog.NewContextCatalogUpdater(p.ContextName)
 	if err != nil {
 		return err
 	}
 	if err := c.Upsert(plugin); err != nil {
 		log.Info("Plugin Info could not be updated in cache")
 	}
+	c.Unlock()
```

Comment:
I think the `c.Unlock()` should be called even if the `Upsert` fails.
Maybe using a `defer` would be simplest, right after the catalog is obtained successfully.

Reply:
It might be good to paraphrase the above as a comment for the benefit of other readers of the code.